app/workers/commit_monitor_handlers/commit_range/gemfile_checker.rb

```diff
@@ -0,0 +1,101 @@
+module CommitMonitorHandlers
+  module CommitRange
+    class GemfileChecker
+      include Sidekiq::Worker
+      sidekiq_options :queue => :miq_bot
+
+      LABEL_NAME = "gem changes".freeze
+
+      def self.handled_branch_modes
+        [:pr]
+      end
+
+      attr_reader :branch, :commits, :github, :pr
+
+      def perform(branch_id, _new_commits)
+        @branch = CommitMonitorBranch.where(:id => branch_id).first
+        
+        if @branch.nil?
+          logger.info("Branch #{branch_id} no longer exists.  Skipping.")
+          return
+        end
+
+        unless repo_enabled?
+          logger.info("#{self.class} only runs in #{enabled_repos}, not #{@branch.repo.fq_name}.  Skipping.")
+          return
+        end
+
+        @pr      = @branch.pr_number
+        @commits = @branch.commits_list
+
+        files = diff_details_for_branch.keys
+        return unless files.any? { |f| File.basename(f) == "Gemfile" }
+
+        process_branch
+      end
+
+      private
+
+      def repo_enabled?
```

Comment:
This seems like something we might want to extract to a common place as we add more an more checkers that need to "configure which repos they want to check".